### PR TITLE
ecl: set library paths explicitly

### PR DESCRIPTION
**Pull request by fjl**
ECL is a Common Lisp to C compiler. When ECL invokes the C compiler, it uses the library directories configured at ECL's build time to set the include path. On Apple Silicon, homebrew has moved to /opt/homebrew, which is not in the default include path of the C compiler. This change makes ECL work when installed in /opt/homebrew by passing those library directories to ./configure.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
This looks like it's `license "GPL-2.0-or-later"`. Can you add in between the `sha256` and `head` lines? CI will need to be re-run anyway, as the ARM runner did not attempt a build. You can just include this change in your existing commit.

**Comment by carlocab** [comment]
Adding `do not merge` until it's re-run with an ARM runner attempting a build. I'll take it off once you've added the license. Thanks again.

**Comment by fjl** [comment]
@carlocab license added

**Comment by fjl** [comment]
The ECL maintainers say it is LGPL-2.1-or-later: https://common-lisp.net/project/ecl/posts/ECL-license.html

**Comment by carlocab** [comment]
Thanks for double-checking the license.

**Comment by fjl** [comment]
CI says it doesn't build. This is expected, because the released version of ECL is not yet compatible with Apple Silicon. I use the `--HEAD` build, and it works for me.

**Comment by fjl** [comment]
However, it looks like they will create a new release soon (next couple days). I will send an upgrade PR then, but do think this change is independent of that. It will also help people with non-default Homebrew prefix.

**Review by carlocab — APPROVED** [review]
Thanks, @fjl.

**Pull request closed without merging**
